sdk/core/ts-http-runtime/src/logger/logger.ts

```diff
@@ -140,10 +140,28 @@ export function createLoggerContext(options: CreateLoggerContextOptions): Logger
     debug.log(...args);
   };
 
+  function contextSetLogLevel(level?: TypeSpecRuntimeLogLevel): void {
+    if (level && !isTypeSpecRuntimeLogLevel(level)) {
+      throw new Error(
+        `Unknown log level '${level}'. Acceptable values: ${TYPESPEC_RUNTIME_LOG_LEVELS.join(",")}`,
+      );
+    }
+    logLevel = level;
+
+    const enabledNamespaces = [];
+    for (const logger of registeredLoggers) {
+      if (shouldEnable(logger)) {
+        enabledNamespaces.push(logger.namespace);
+      }
+    }
+
+    debug.enable(enabledNamespaces.join(","));
+  }
+
   if (logLevelFromEnv) {
     // avoid calling setLogLevel because we don't want a mis-set environment variable to crash
     if (isTypeSpecRuntimeLogLevel(logLevelFromEnv)) {
```

Comment:
should we ditch this check since it is already being done inside `contextSetLogLevel`? Also, why sometime we throw an error and in some other times use `console.error`?

Reply:
This is the pre-existing behavior, and to me it seems reasonable. If `setLogLevel` is called programatically with an invalid log level (which shouldn't be possible in TypeScript, but could happen in JS), throwing an error seems reasonable to me. On the other hand, I don't think an incorrectly set environment variable should crash the program, which is what would happen if we threw here. I suppose we could catch the error here and then `console.error` in the catch instead of having this guard, but it has the same effect at the end of the day, and we'd have to do a cast instead of using the type guard.

Reply:
Runtime behavior is fine. I just don't like that we're checking the same thing twice.